### ISSUE TITLE
[BUG] Media player buttons lag while media is playing in Safari

### ISSUE THREAD

**Issue by elynema** [samvera-labs/ramp#338]
# Description

Clicking on buttons within the media player, or using the progress bar to skip ahead in the media item, results in extreme lag in Safari. At first it may respond within reason, but if you click several times in the progress bar, the lag gets beyond 5 seconds, and while it is lagging, other buttons do not work (ex: pause button). I am able to replicate these issues on the avalon-dev media object and playlist page, as well as the Ramp demo site, so I think it's a Ramp issue.

This doesn't seem to happen if the media item is not actually playing (before you start playback). These issues do not happen in Chrome or Firefox.

Eventually, Safari sends a warning that the webpage is using significant memory. See an example of this message from the Ramp demo site in Safari:

<img width="1238" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNUc5QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--6fba1472dfcdd7cd31fbda990f6cb5abd292ec7c/image.png" alt="image.png" />

In the attached video, you can see that I can click in the progress bar to jump around before starting to play the video. Once playback is started, the first click in the progress bar works slowly, the second click even more slowly, and by the third click (back toward the beginning of the video), it just doesn't respond. After that, clicking the pause button also doesn't work and you can see the current time display for the video is stuck and not advancing, even though the video is playing.
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNTY5QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--72f49531158198bd9044c2e83e90f1004721bb29/Screen%20Recording%202024-01-11%20at%202.45.06%20PM.mov)

**Comment by joncameron** [comment]
React hooks would be the first place to check (state updates).

**Comment by elynema** [comment]
This was working ok in Safari on my MacOS laptop. However, Safari on iPad is having issues. After scrubbing back and forth, the page eventually crashes.

Safari on iPhone works fine since it opens in native player.

@Dananji Do you want me to create a new issue for that? Do you think that is likely to be a different solution?

**Comment by Dananji** [comment]
It could be due to the fact that, Ramp's progress bar not having any touch event handlers at the moment. 
It seems, the behavior you are describing is specific to iPads while this ticket was focused on desktop browsers for MacOS.

If it's the touch event handlers then, it is definitely a different solution which I think can be a part of #354, therefore adding a note or updating the done looks like in #354 should be enough.

**Comment by elynema** [comment]
I added a note to #354, which already has pretty comprehensive "Done looks like."
